docs/variant_merging.md

```diff
@@ -1,3 +1,227 @@
+
+```
+## <headers omitted>
+#CHROM  POS   ID    REF   ALT   QUAL  FILTER  INFO         FORMAT  S3
+1       100   rs1   A     T     7     q10     DP=11;AA=G   GT:GQ   1/1:2
+```
+
+If you load both files together (i.e. by specifying
+`--input_pattern gs://mybucket/*.vcf`), you would end up with 2 rows in
+BigQuery. One with calls `S1` and `S2`, and another with just call `S3` even
+though all calls have the same variant.
+
+Variant merging can be used to merge calls across files. There are different
+merging strategies and you can specify any of the following
+strategies by adding `--variant_merge_strategy <merge_strategy>` to the
+arguments.
+
+## MOVE_TO_CALLS strategy
+
+This strategy merges calls that have the same `reference_name`,
+`start_position`, `end_position`, `reference_bases`, and all `alternate_bases`
+across files. By default, the variant fields are merged as follows:
+
+* `Calls` are merged together in no particular order.
+* The highest `quality` value is chosen among all variants being merged.
+* All `filter` values are merged and deduplicated.
+* All `names` (`ID` column in the VCF spec) are merged and deduplicated.
+* For each `INFO` field, one is chosen as representative for the *entire* merged
+  variant record in *no particular order*.
+
+Depending on your use case, you may customize the merge strategy as follows:
+
+* Adding `--copy_quality_to_calls` and/or `--copy_filter_to_calls` to the
+  arguments will copy the `quality` and/or `filter` values in each file to the
+  set of calls specified in that file. This option is useful when merging
+  single-call VCF files as `quality` and `filter` values usually correspond
+  the call-level quality/filter. Note that the variant-level `quality` and
+  `filter` values will still be merged according to the above logic.
+* Adding `--info_keys_to_move_to_calls_regex <regex>` will move `INFO` fields
+  that match the provided regular expression to the associated calls. This is
+  useful to ensure all `INFO` fields are kept when merging variants.
+
+The examples below help with illustrating the details of the above options.
+You may also read the code and documentation in
+[move_to_calls_strategy.py](../gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py).
+Particularly the `get_merged_variants` method implements the merging logic.
+
+### Examples
+
+The following examples illustrate the merging outcome using the example files
+`gs://mybucket/a.vcf` and `gs://mybucket/b.vcf` that are defined above.
+
+* Default (i.e. just adding `--variant_merge_strategy MOVE_TO_CALLS`):
+
+```
+reference_name: "1"
+start_position: 100
+end_position: 101
+reference_bases: "A"
+alternate_bases: {alt: "T"}
+names: ["rs1"]
+quality: 29
+filter: ["PASS", "q10"]
+call: {
+  [
+    name: "S3",
+    genotype: [1, 1]
+    GQ: 2
+  ],
+  [
+    name: "S1",
+    genotype: [0, 1]
+    phaseset: "*"
+    GQ: 48
+  ],
+  [
+    name: "S2",
+    genotype: [0, 0]
+    phaseset: null
```

Comment:
Do we care pointing out why phaseset is null in S2 but is missing in S3?

Reply:
ah, good catch! It should be null everywhere :)